src/GovUk.Education.ExploreEducationStatistics.Admin/Services/UserRoleService.cs

```diff
@@ -0,0 +1,307 @@
+
+        public UserRoleService(UsersAndRolesDbContext usersAndRolesDbContext,
+            ContentDbContext contentDbContext,
+            IPersistenceHelper<ContentDbContext> contentPersistenceHelper,
+            IPersistenceHelper<UsersAndRolesDbContext> usersAndRolesPersistenceHelper,
+            IEmailTemplateService emailTemplateService,
+            IUserService userService,
+            IUserPublicationRoleRepository userPublicationRoleRepository,
+            IUserReleaseRoleRepository userReleaseRoleRepository,
+            UserManager<ApplicationUser> userManager)
+        {
+            _usersAndRolesDbContext = usersAndRolesDbContext;
+            _contentDbContext = contentDbContext;
+            _contentPersistenceHelper = contentPersistenceHelper;
+            _usersAndRolesPersistenceHelper = usersAndRolesPersistenceHelper;
+            _emailTemplateService = emailTemplateService;
+            _userService = userService;
+            _userPublicationRoleRepository = userPublicationRoleRepository;
+            _userReleaseRoleRepository = userReleaseRoleRepository;
+            _userManager = userManager;
+        }
+
+        public async Task<Either<ActionResult, Unit>> AddGlobalRole(string userId, string roleId)
+        {
+            return await _userService
+                .CheckCanManageAllUsers()
+                .OnSuccess(async () =>
+                {
+                    return await _usersAndRolesPersistenceHelper
+                        .CheckEntityExists<ApplicationUser, string>(userId)
+                        .OnSuccessCombineWith(user =>_usersAndRolesPersistenceHelper
+                            .CheckEntityExists<IdentityRole, string>(roleId))
+                        .OnSuccessVoid(async tuple => 
+                        {
+                            var (user, role) = tuple;
+                            await _userManager.AddToRoleAsync(user, role.Name);
+                        });
+                });
+        }
+
+        public async Task<Either<ActionResult, Unit>> AddPublicationRole(Guid userId, Guid publicationId, PublicationRole role)
+        {
+            return await _userService
+                .CheckCanManageAllUsers()
+                .OnSuccess(async () =>
+                {
+                    return await _usersAndRolesPersistenceHelper
+                        .CheckEntityExists<ApplicationUser, string>(userId.ToString())
+                        .OnSuccessCombineWith(user => _contentPersistenceHelper.CheckEntityExists<Publication>(publicationId))
+                        .OnSuccessCombineWith(release => ValidatePublicationRoleCanBeAdded(userId, publicationId, role))
+                        .OnSuccessVoid(async tuple =>
+                        {
+                            var (userAndPublication, _) = tuple;
+                            var (user, publication) = userAndPublication;
+                            await _userPublicationRoleRepository.Create(
```

Comment:
If all the parameters in `UserPublicationRoleRepository.Create()` are non-optional, I don't think you gain anything by explicitly naming parameters here as you're passing them in?  E.g. `userId: userId` can probably just be `userId`?

Reply:
There are a few more instances of this elsewhere in the PR too.

Reply:
This is something Nick suggested we introduce a while ago and I've been doing in places, especially where there's multiple parameters of the same type. It helps with refactoring where the order of parameters is swapped, as then the calling code doesn't necessarily need updating since the params are named. Where multiple parameters are of the same type like Guid here, it helps seeing them in the IDE to spot mistakes.